aya-obj/src/btf/btf.rs

```diff
@@ -821,6 +899,139 @@ impl Default for Btf {
 }
 
 impl Object {
+    fn patch_extern_data_internal(
```

Comment:
this name is confusing, maybe prepare_kconfig_section or something?

even the patch_extern_data name below is confusing. Patch makes me think you're
relocating but you're not, you're patching _the symbols_ and preparing the
section. 

Not sure what the best name would be.